### PR TITLE
[SPARK-14396] [SQL] Throw Exceptions for DDLs of Partitioned Views

### PR DESCRIPTION
**Pull request by gatorsmile**
#### What changes were proposed in this pull request?

Because the concept of partitioning is associated with physical tables, we disable all the supports of partitioned views, which are defined in the following three commands in [Hive DDL Manual](https://cwiki.apache.org/confluence/display/Hive/LanguageManual+DDL#LanguageManualDDL-Create/Drop/AlterView):

```
ALTER VIEW view DROP [IF EXISTS] PARTITION spec1[, PARTITION spec2, ...];

ALTER VIEW view ADD [IF NOT EXISTS] PARTITION spec;

CREATE VIEW [IF NOT EXISTS] [db_name.]view_name [(column_name [COMMENT column_comment], ...) ]
  [COMMENT view_comment]
  [TBLPROPERTIES (property_name = property_value, ...)]
  AS SELECT ...;
```

An exception is thrown when users issue any of these three DDL commands.
#### How was this patch tested?

Added test cases for parsing create view and changed the existing test cases to verify if the exceptions are thrown.

### CODE REVIEW

File: sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4 [hunk 1]
```diff
@@ -75,17 +75,17 @@ statement
     | ALTER TABLE tableIdentifier NOT STORED AS DIRECTORIES            #unstoreTable
     | ALTER TABLE tableIdentifier
         SET SKEWED LOCATION skewedLocationList                         #setTableSkewLocations
-    | ALTER TABLE tableIdentifier ADD (IF NOT EXISTS)?
+    | ALTER kind=TABLE tableIdentifier ADD (IF NOT EXISTS)?
```
Comment (hvanhovell) [inline sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4]: Minor: we don't really need kind. We can also just check for the existance of the VIEW keyword, e.g.: `ctx.VIEW != null`.

Comment (gatorsmile) [inline sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4]: Thanks! Will do.

**Comment by gatorsmile** [comment]
Thanks for the review! @hvanhovell 

Also cc @yhuai @andrewor14

File: sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala [hunk 2]
```diff
@@ -553,9 +553,13 @@ class SparkSqlAstBuilder extends AstBuilder {
    *   ALTER TABLE table ADD [IF NOT EXISTS] PARTITION spec [LOCATION 'loc1']
    *   ALTER VIEW view ADD [IF NOT EXISTS] PARTITION spec
    * }}}
+   *
+   * ALTER VIEW ... DROP PARTITION ... is not supported because the concept of partitioning
+   * is associated with physical tables
    */
   override def visitAddTablePartition(
       ctx: AddTablePartitionContext): LogicalPlan = withOrigin(ctx) {
+    if (ctx.VIEW != null) throw new ParseException(s"Unsupported partitioned view", ctx)
```
Comment (andrewor14) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala]: I think @yhuai prefers to say something like `Operation not allowed: partitioned views` for the message

**Comment by andrewor14** [comment]
Looks pretty good, just a minor comment.

File: sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala (hunk 2 above)
Comment (gatorsmile) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala]: Will do it now. Thanks!

**Comment by gatorsmile** [comment]
retest this please

**Comment by andrewor14** [comment]
Build is broken I think. retest this please

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/HiveSqlParser.scala [hunk 3]
```diff
@@ -215,11 +215,20 @@ class HiveSqlAstBuilder extends SparkSqlAstBuilder {
...
+   * {{{
+   *   CREATE VIEW [IF NOT EXISTS] [db_name.]view_name
+   *   [(column_name [COMMENT column_comment], ...) ]
+   *   [COMMENT view_comment]
+   *   [TBLPROPERTIES (property_name = property_value, ...)]
+   *   AS SELECT ...;
+   * }}}
    */
   override def visitCreateView(ctx: CreateViewContext): LogicalPlan = withOrigin(ctx) {
     // Pass a partitioned view on to hive.
```
Comment (andrewor14) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/HiveSqlParser.scala]: this needs to be deleted

**Comment by andrewor14** [comment]
LGTM we just need to make tests pass now

**Comment by andrewor14** [comment]
Merged into master.

**Comment by yhuai** [comment]
@gatorsmile Seems scala 2.10 build is broken. Can you take a look?

**Comment by gatorsmile** [comment]
Sure. Could you show me how to find the build log?

**Comment by gatorsmile** [comment]
Or I need to do it in my local environment?

**Comment by gatorsmile** [comment]
https://amplab.cs.berkeley.edu/jenkins/job/spark-master-compile-maven-scala-2.10/ ?

**Comment by gatorsmile** [comment]
I see the problem. 

```
[error] /home/jenkins/workspace/spark-master-compile-maven-scala-2.10/sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveDDLCommandSuite.scala:266: value contains is not a member of Option[String]
[error]     assert(desc.viewText.contains("SELECT * FROM tab1"))
[error]                          ^
[error] /home/jenkins/workspace/spark-master-compile-maven-scala-2.10/sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveDDLCommandSuite.scala:267: value contains is not a member of Option[String]
[error]     assert(desc.viewOriginalText.contains("SELECT * FROM tab1"))
[error]                                  ^
[error] /home/jenkins/workspace/spark-master-compile-maven-scala-2.10/sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveDDLCommandSuite.scala:293: value contains is not a member of Option[String]
[error]     assert(desc.viewText.contains("SELECT * FROM tab1"))
[error]                          ^
[error] /home/jenkins/workspace/spark-master-compile-maven-scala-2.10/sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveDDLCommandSuite.scala:294: value contains is not a member of Option[String]
[error]     assert(desc.viewOriginalText.contains("SELECT * FROM tab1"))
[error]                                  ^
[error] four errors found
[error] Compile failed at Apr 5, 2016 10:59:09 PM [10.502s]
```

**Comment by gatorsmile** [comment]
Will fix it using a PR with [HOT]. Sorry for it.

**Pull request closed without merging**